build_tools/gradle/build_android.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Cross-compile the IREE's TFLite Java Bindings with Gradle and CMake.
+
+set -x
+set -e
+set -o pipefail
+
+# --------------------------------------------------------------------------- #
+ROOT_DIR=$(git rev-parse --show-toplevel)
+
+CMAKE_BIN=${CMAKE_BIN:-$(which cmake)}
+
+"${CMAKE_BIN?}" --version
+ninja --version
+gradle --version
+
+cd ${ROOT_DIR?}
+
+# --------------------------------------------------------------------------- #
+# Build the host libraries
+
+HOST_BUILD_DIR=${ROOT_DIR?}/build-host
+HOST_INSTALL_DIR=${HOST_BUILD_DIR}/install
+
+cmake -G Ninja -B ${HOST_BUILD_DIR} \
+-DIREE_TARGET_BACKENDS_TO_BUILD=vmvx \
+-DIREE_HAL_DRIVERS_TO_BUILD=vmvx \
+-DIREE_BUILD_COMPILER=OFF \
+-DIREE_BUILD_TESTS=OFF \
+-DIREE_BUILD_SAMPLES=OFF \
+-DCMAKE_C_COMPILER=clang \
+-DCMAKE_CXX_COMPILER=clang++ \
+-DCMAKE_INSTALL_PREFIX=${HOST_INSTALL_DIR} .
+
+cmake --build ${HOST_BUILD_DIR} --target install
+
+# --------------------------------------------------------------------------- #
+# Build native libraries
+
+ANDROID_BUILD_DIR=${ROOT_DIR?}/build-android
+
+# Todo: Support multiple ABIs. For now we build a singe abi: arm64.
+ANDROID_ABI=arm64-v8a
+
+cmake -G Ninja -B ${ANDROID_BUILD_DIR} \
+-DCMAKE_TOOLCHAIN_FILE="${ANDROID_NDK}/build/cmake/android.toolchain.cmake" \
+-DIREE_HOST_BINARY_ROOT="${HOST_INSTALL_DIR}" \
+-DANDROID_ABI="${ANDROID_ABI}" \
+-DANDROID_PLATFORM="android-29" \
```

Comment:
This will need to change in some way (I'm not sure what our options are yet) if we are distributing binaries.

Reply:
Sorry, I don't quite follow this comment. Can you elaborate? What needs to change and why? (android version 29)??

Reply:
We shouldn't restrict users of our library to a single Android version. If this is a "maximum supported" version then that's okay, if it's only one version then we should see what our options are for publishing one per version, etc.

Reply:
I see. Right now our `build.gradle` is configured to target api 29 with api 28 specified as the minimum. Its likely the library will actually support other version (and we can lower the min target) but this will require more looking into/testing. However, since our library is compiled with version 29, `-DANDROID_PLATFORM=` must match that (there is nothing to change here atm). 

More details on `compileSdkVersion`: https://developer.android.com/studio/build#module-level

I can list a todo for multi-version support for now.